From 078fb2057dacd1ccdbbb73dc70605b6f6059f194 Mon Sep 17 00:00:00 2001
From: Petr Oros <poros@redhat.com>
Date: Tue, 24 Jun 2014 08:56:08 -0400
Subject: [Documentation] fix incorrect formula to calculate CommitLimit value

Message-id: <1403600168.2913.7.camel@holly>
Patchwork-id: 84665
O-Subject: [PATCH RHEL7 BZ1060570] doc: fix incorrect formula to calculate CommitLimit value
Bugzilla: 1060570
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Rafael Aquini <aquini@redhat.com>

Upstream commit: 7a9e6da11c9478741947de3a37ac7de26abd23f0
Bz: https://bugzilla.redhat.com/show_bug.cgi?id=1060570

Author: Petr Oros <poros@redhat.com>
Date: Thu, 22 May 2014 14:04:44 +0200

Description:
 The formula to calculate "CommitLimit" value mentioned in kernel documentation is incorrect.
 Right formula is: CommitLimit = ([total RAM pages] - [total huge TLB pages]) * overcommit_ratio / 100 + [total swap pages]

 Signed-off-by: Petr Oros <poros@redhat.com>
 Acked-by: Randy Dunlap <rdunlap@infradead.org>
 Signed-off-by: Jiri Kosina <jkosina@suse.cz>

Signed-off-by: Petr Oros <poros@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/Documentation/filesystems/proc.txt b/Documentation/filesystems/proc.txt
index 2b68371..159deb0 100644
--- a/Documentation/filesystems/proc.txt
+++ b/Documentation/filesystems/proc.txt
@@ -848,7 +848,8 @@ WritebackTmp: Memory used by FUSE for temporary writeback buffers
               if strict overcommit accounting is enabled (mode 2 in
               'vm.overcommit_memory').
               The CommitLimit is calculated with the following formula:
-              CommitLimit = ('vm.overcommit_ratio' * Physical RAM) + Swap
+              CommitLimit = ([total RAM pages] - [total huge TLB pages]) *
+                             overcommit_ratio / 100 + [total swap pages]
               For example, on a system with 1G of physical RAM and 7G
               of swap with a `vm.overcommit_ratio` of 30 it would
               yield a CommitLimit of 7.3G.
-- 
1.7.1